File: src/main/java/com/turchinsky/entities/ScheduleDetailsEntity.java
package com.turchinsky.entities;


import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "schedule_details")
public class ScheduleDetailsEntity {

    @Basic
    @Column(name = "route_id")
    private int routeId;

    @Basic
    @Column(name = "station_id")
    private int stationId;

    @Basic
    @Column(name = "station_order")
    private int stationOrder;

    @Basic
    @Column(name = "train_id")
    private int trainId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "line_id"/*, nullable = false, insertable = false, updatable = false*/)
    private int lineId;

    @Basic
    @Column(name = "arrival_time")
    private Timestamp arrivalTime;

    @Basic
    @Column(name = "departure_time")
    private Timestamp departureTime;

    @Basic
    @Column(name = "free_seats")
    private int freeSeats;

    @Basic
    @Column(name = "ordered_seats")
    private int orderedSeats;

    @Basic
    @Column(name = "routegroupid")
    private int routeGroupId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumns({
                         @JoinColumn(name = "route_id", referencedColumnName = "route_id", updatable = false,
                                     insertable = false),
                         @JoinColumn(name = "station_id", referencedColumnName = "station_id", updatable = false,
                                     insertable = false),
                         @JoinColumn(name = "station_order", referencedColumnName = "station_order", updatable = false,
                                     insertable = false)}
    )
    private RouteDetailsEntity refRouteDetailsEntity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "train_id", referencedColumnName = "id", updatable = false, insertable = false)
    private TrainEntity refTrainEntity;


    public ScheduleDetailsEntity() {
    }

    public ScheduleDetailsEntity(int routeGroupId, int stationId, int stationOrder, int trainId,
                                 Timestamp arrivalTime, Timestamp departureTime, int freeSeats, int orderedSeats,
                                 int routeId) {
        this.routeId = routeId;
        this.stationId = stationId;
        this.stationOrder = stationOrder;
        this.trainId = trainId;
        this.arrivalTime = arrivalTime;
        this.departureTime = departureTime;
        this.freeSeats = freeSeats;
        this.orderedSeats = orderedSeats;
        this.routeGroupId = routeGroupId;
    }

    //    @OneToMany(mappedBy = "refScheduleDetailsEntity")
//    private List<TicketEntity> refTicketEntities;


    public int getRouteId() {
        return routeId;
    }

    public void setRouteId(int routeId) {
        this.routeId = routeId;
    }

    public int getStationId() {
        return stationId;
    }

    public void setStationId(int stationId) {
        this.stationId = stationId;
    }

    public int getStationOrder() {
        return stationOrder;
    }

    public void setStationOrder(int stationOrder) {
        this.stationOrder = stationOrder;
    }

    public int getTrainId() {
        return trainId;
    }

    public void setTrainId(int trainId) {
        this.trainId = trainId;
    }

    public int getLineId() {
        return lineId;
    }

    public void setLineId(int lineId) {
        this.lineId = lineId;
    }

    public Timestamp getArrivalTime() {
        return arrivalTime;
    }

    public void setArrivalTime(Timestamp arrivalTime) {
        this.arrivalTime = arrivalTime;
    }

    public Timestamp getDepartureTime() {
        return departureTime;
    }

    public void setDepartureTime(Timestamp departureTime) {
        this.departureTime = departureTime;
    }

    public int getFreeSeats() {
        return freeSeats;
    }

    public void setFreeSeats(int freeSeats) {
        this.freeSeats = freeSeats;
    }

    public int getOrderedSeats() {
        return orderedSeats;
    }

    public void setOrderedSeats(int orderedSeats) {
        this.orderedSeats = orderedSeats;
    }

    public RouteDetailsEntity getRefRouteDetailsEntity() {
        return refRouteDetailsEntity;
    }

    public void setRefRouteDetailsEntity(RouteDetailsEntity refRouteDetailsEntity) {
        this.refRouteDetailsEntity = refRouteDetailsEntity;
    }

    public int getRouteGroupId() {
        return routeGroupId;
    }

    public void setRouteGroupId(int groupId) {
        this.routeGroupId = groupId;
    }

    public TrainEntity getRefTrainEntity() {
        return refTrainEntity;
    }

    public void setRefTrainEntity(TrainEntity refTrainEntity) {
        this.refTrainEntity = refTrainEntity;
    }

//    public List<TicketEntity> getRefTicketEntities() {
//        return refTicketEntities;
//    }
//
//    public void setRefTicketEntities(List<TicketEntity> refTicketEntities) {
//        this.refTicketEntities = refTicketEntities;
//    }


    @Override public String toString() {
        return "ScheduleDetailsEntity{" +
                "routeId=" + routeId +
                ", stationId=" + stationId +
                ", stationOrder=" + stationOrder +
                ", trainId=" + trainId +
                ", lineId=" + lineId +
                ", arrivalTime=" + arrivalTime +
                ", departureTime=" + departureTime +
                ", freeSeats=" + freeSeats +
                ", orderedSeats=" + orderedSeats +
                ", routeGroupId=" + routeGroupId +
                '}';
    }

}

File: src/main/java/com/turchinsky/kt/ScheduleDetailsEntity.kt
//package com.turchinsky.kt
//
//import javax.persistence.*
//
//@Entity
//@Table(name = "schedule_details", schema = "public", catalog = "sbb")
//open class ScheduleDetailsEntity {
//    @get:Basic
//    @get:Column(name = "route_id", nullable = true, insertable = false, updatable = false)
//    var routeId: Int? = null
//
//    @get:Basic
//    @get:Column(name = "station_id", nullable = true, insertable = false, updatable = false)
//    var stationId: Int? = null
//
//    @get:Basic
//    @get:Column(name = "station_order", nullable = true, insertable = false, updatable = false)
//    var stationOrder: Int? = null
//
//    @get:Basic
//    @get:Column(name = "train_id", nullable = true, insertable = false, updatable = false)
//    var trainId: Int? = null
//
//    @get:Id
//    @get:Column(name = "line_id", nullable = false, insertable = false, updatable = false)
//    var lineId: Int? = null
//
//    @get:Basic
//    @get:Column(name = "arrival_time", nullable = true)
//    var arrivalTime: java.sql.Timestamp? = null
//
//    @get:Basic
//    @get:Column(name = "departure_time", nullable = true)
//    var departureTime: java.sql.Timestamp? = null
//
//    @get:Basic
//    @get:Column(name = "free_seats", nullable = true)
//    var freeSeats: Int? = null
//
//    @get:Basic
//    @get:Column(name = "ordered_seats", nullable = true)
//    var orderedSeats: Int? = null
//
//    @get:ManyToOne(fetch = FetchType.LAZY)
//    @get:JoinColumns(
//            JoinColumn(name = "route_id", referencedColumnName = "route_id"),
//            JoinColumn(name = "station_id", referencedColumnName = "station_id"),
//            JoinColumn(name = "station_order", referencedColumnName = "station_order")
//    )
//    var refRouteDetailsEntity: RouteDetailsEntity? = null
//
//    @get:ManyToOne(fetch = FetchType.LAZY)
//    @get:JoinColumn(name = "train_id", referencedColumnName = "id")
//    var refTrainEntity: TrainEntity? = null
//
//    @get:OneToMany(mappedBy = "refScheduleDetailsEntity")
//    var refTicketEntities: List<TicketEntity>? = null
//
//    override fun toString(): String =
//            "Entity of type: ${javaClass.name} ( " +
//                    "routeId = $routeId " +
//                    "stationId = $stationId " +
//                    "stationOrder = $stationOrder " +
//                    "trainId = $trainId " +
//                    "lineId = $lineId " +
//                    "arrivalTime = $arrivalTime " +
//                    "departureTime = $departureTime " +
//                    "freeSeats = $freeSeats " +
//                    "orderedSeats = $orderedSeats " +
//                    ")"
//
//    // constant value returned to avoid entity inequality to itself before and after it's update/merge
//    override fun hashCode(): Int = 42
//
//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//        other as ScheduleDetailsEntity
//
//        if (routeId != other.routeId) return false
//        if (stationId != other.stationId) return false
//        if (stationOrder != other.stationOrder) return false
//        if (trainId != other.trainId) return false
//        if (lineId != other.lineId) return false
//        if (arrivalTime != other.arrivalTime) return false
//        if (departureTime != other.departureTime) return false
//        if (freeSeats != other.freeSeats) return false
//        if (orderedSeats != other.orderedSeats) return false
//
//        return true
//    }
//
//}
//
